frontend/src/components/dashboard/home/VisitsDensityWidget.vue

```diff
@@ -0,0 +1,216 @@
+                            lines: {
+                                show: false
+                            }
+                        },
+                        padding: {
+                            left: 15
+                        },
+                    },
+                    noData: {
+                        text: 'No users for this period',
+                        style: {
+                            color: '#88929a',
+                            fontSize: '18px'
+                        }
+                    },
+                    legend: {
+                        show: false
+                    },
+                    plotOptions: {
+                        heatmap: {
+                            radius: 6,
+                        }
+                    },
+                    states: {
+                        hover: {
+                            filter: {
+                                type: 'none',
+                            }
+                        },
+                        active: {
+                            filter: {
+                                type: 'none',
+                            }
+                        },
+                    },
+                    tooltip: {
+                        enabled: true,
+                        custom: ({series, seriesIndex, dataPointIndex}) => {
+                            return `<span style="padding: 10px">${series[seriesIndex][dataPointIndex]}</span>`;
+                        }
+                    },
+                    xaxis: {
+                        labels: {
+                            show: true
+                        },
+                        axisBorder: {
+                            show: false,
+                        },
+                        axisTicks: {
+                            show: false,
+                        },
+                        tooltip: {
+                            enabled: false
+                        }
+                    },
+                    yaxis: {
+                        opposite: true,
+                        reversed: true,
+                        labels: {
+                            align: 'right'
+                        }
+                    }
+                }
+            };
+        },
+        computed: {
+            ...mapGetters('visits_density_widget', {
+                getSelectedPeriod: GET_SELECTED_PERIOD,
+                getVisitsData: GET_VISITS_DATA,
+                isFetching: IS_FETCHING
+            }),
+        },
+        created() {
+            this.fetchWidgetData();
+            this.drawHeatmap();
+        },
+        methods: {
+            ...mapActions('visits_density_widget', {
+                changeSelectedPeriod: CHANGE_SELECTED_PERIOD,
+                fetchWidgetData: FETCH_WIDGET_DATA
+            }),
+            drawHeatmap () {
+                if (_.isEmpty(this.getVisitsData)) {
+                    this.chartOptions.xaxis.labels.show = false;
```

Comment:
Not quite sure that method, which is responsible for providing data, should do anything with local state (at least it seems this method is used just as data provider)